kernel/log.h

```diff
@@ -69,6 +71,22 @@ YOSYS_NAMESPACE_BEGIN
 #define S__LINE__sub1(x) S__LINE__sub2(x)
 #define S__LINE__ S__LINE__sub1(__LINE__)
 
+#if defined(_MSC_VER)
+# define YS_DEBUGTRAP __debugbreak()
+#elif defined(__unix__)
+# if defined(__clang__)
+#  if defined(__builtin_debugtrap)
+#   define YS_DEBUGTRAP __builtin_debugtrap()
+#  else
+#   define YS_DEBUGTRAP do{if(ptrace(PTRACE_TRACEME, 0, nullptr, 0) == -1){ raise(SIGTRAP); }} while(0);
```

Comment:
Style nit: if the definitions that expand into functions don't end with a `;`, then the defines that expand into do-while loops shouldn't either.